src/app/pages/RewardPage/hooks/useGetNextPositiveCheckpoint.ts

```diff
@@ -0,0 +1,62 @@
+import { useAccount } from 'app/hooks/useAccount';
+import { useCallback, useEffect, useState } from 'react';
+import { contractReader } from 'utils/sovryn/contract-reader';
+
+const MAX_NEXT_POSITIVE_CHECKPOINT = 75;
+
+type UserCheckpoint = {
+  checkpointNum: string;
+  hasFees: boolean;
+  hasSkippedCheckpoints: boolean;
+};
+
+export const useGetNextPositiveCheckpoint = (
+  contractAddress: string,
+  totalTokenCheckpoints: number,
+) => {
+  const address = useAccount();
+  const [userCheckpoint, setUserCheckpoint] = useState<UserCheckpoint>();
+
+  const updateNextPositiveCheckpoint = useCallback(async () => {
+    let index = 1;
+
+    while (index < 5) {
+      const checkpoint = await contractReader.call<UserCheckpoint>(
+        'feeSharingProxy',
+        'getNextPositiveUserCheckpoint',
+        [
+          address,
+          contractAddress,
+          MAX_NEXT_POSITIVE_CHECKPOINT * (index - 1),
+          MAX_NEXT_POSITIVE_CHECKPOINT * index,
+        ],
+      );
+
+      if (
+        !!checkpoint.hasFees ||
+        totalTokenCheckpoints < MAX_NEXT_POSITIVE_CHECKPOINT * index
+      ) {
+        return setUserCheckpoint({
+          checkpointNum: checkpoint.checkpointNum,
+          hasFees: checkpoint.hasFees,
+          hasSkippedCheckpoints: checkpoint.hasSkippedCheckpoints,
+        });
```

Comment:
```suggestion
     return setUserCheckpoint({ ...checkpoint });
```

Reply:
That would be fine, the props you need are there and you don't mind extra props. But if you're worried about that, you can always do this:
`const { checkpointNum, hasFees, hasSkippedCheckpoints } = checkpoint;

        return setUserCheckpoint({
          checkpointNum,
          hasFees,
          hasSkippedCheckpoints,
        });
        `